integration_tests/conftest.py

```diff
@@ -1,21 +1,47 @@
 import pytest
 from cloudpathlib import S3Path
 import os
+import boto3
 
+PIPELINE_BUCKET = os.environ.get("INGEST_PIPELINE_BUCKET")
+S3_CLIENT = boto3.client("s3")
 
-def get_bucket_files_with_suffix(bucket: S3Path, suffix: str) -> list[S3Path]:
-    """Get all the files in a bucket with a given suffix."""
-    bucket_files = []
-    for pattern in ["*", "*/*", "*/*/*", "*/*/*/*"]:
-        files = list(bucket.glob(pattern + suffix))
-        bucket_files.extend(set(files))
-    return bucket_files
+
+def get_bucket_files_with_suffix(bucket: str, suffix: str) -> list[S3Path]:
+    """Retrieve all the files in an s3 bucket with a given suffix."""
+    response = S3_CLIENT.list_objects_v2(Bucket=bucket)
+
+    if "Contents" in response:
+        files_with_suffix = [
+            obj["Key"] for obj in response["Contents"] if obj["Key"].endswith(suffix)
+        ]
+
+        # If there are more than 1000 objects continue listing
+        while response["IsTruncated"]:
+            response = S3_CLIENT.list_objects_v2(
+                Bucket=bucket, ContinuationToken=response["NextContinuationToken"]
+            )
+            files_with_suffix.extend(
+                [
+                    obj["Key"]
+                    for obj in response["Contents"]
+                    if obj["Key"].endswith(suffix)
+                ]
+            )
+
+        # Convert to s3 paths and return
+        return [
+            S3Path(os.path.join("s3://", bucket, file)) for file in files_with_suffix
+        ]
+    return []
```

Comment:
There might be a cleaner approach to getting all the objects with the paginator if your interested, this snippet would get most of the way, you might be able to repurpose that search method to search the suffix as well: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#filter-objects-by-last-modified-time-using-jmespath